node/txsimulator/txSimulator.go

```diff
@@ -0,0 +1,53 @@
+package txsimulator
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/node"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type transactionSimulator struct {
+	txProcessor TransactionProcessor
+}
+
+// New returns a new instance of a transactionSimulator
+func New(txProcessor TransactionProcessor) (*transactionSimulator, error) {
+	if check.IfNil(txProcessor) {
+		return nil, node.ErrNilTxSimulatorProcessor
+	}
+
+	return &transactionSimulator{
+		txProcessor: txProcessor,
+	}, nil
+}
+
+// ProcessTx will process the transaction in a special environment, where state-writing is not allowed
+func (ts *transactionSimulator) ProcessTx(accounts state.AccountsAdapter, tx *transaction.Transaction) (*transaction.SimulationResults, error) {
+	newAccounts, err := NewReadOnlyAccountsDB(accounts)
```

Comment:
you can create this only once - at constructor level. and you have access to all the current state. no need to recreate for every simulation process.

Reply:
done